internal/hardware/validator.go

```diff
@@ -20,12 +20,13 @@ import (
 //go:generate mockgen -source=validator.go -package=hardware -destination=mock_validator.go
 type Validator interface {
 	GetHostValidDisks(host *models.Host) ([]*models.Disk, error)
-	GetHostRequirements(role models.HostRole) models.HostRequirementsRole
+	GetHostRequirements() *models.VersionedHostRequirements
 	GetHostInstallationPath(host *models.Host) string
 	GetClusterHostRequirements(ctx context.Context, cluster *common.Cluster, host *models.Host) (*models.ClusterHostRequirements, error)
 	DiskIsEligible(disk *models.Disk) []string
 	ListEligibleDisks(inventory *models.Inventory) []*models.Disk
 	GetInstallationDiskSpeedThresholdMs() int64
+	GetPreflightHardwareRequirements(ctx context.Context, cluster *common.Cluster) (*models.PreflightHardwareRequirements, error)
```

Comment:
please add a comment about this method - not sure I understood from the name what is the difference between this method and GetHostRequirements. 
I think that it's worth mentioning if it takes into account the operators requirements

Reply:
Added. 

`GetPreflightHardwareRequirements` provides hardware (host) requirements that can be calculated only using cluster information. Returned information describe requirements coming from OCP and OLM operators. It should replace GetHostRequirements when its endpoint is not used anymore.